functorch/test/test_eager_transforms.py

```diff
@@ -47,145 +47,106 @@
+        res = _slice_argnums(args, 0)
         self.assertEqual(res, (0.1,))
-        self.assertEqual(f_new(*res), 0.1)
 
-        f_new, res = _argnums_partial(f, args, 4)
+        res = _slice_argnums(args, 4)
         self.assertEqual(res, (4.1,))
-        self.assertEqual(f_new(*res), 0.1)
 
     def test_flat_args_with_negative_int_argnum(self):
         args = (0.1, 1.1, 2.1, 3.1, 4.1)
 
-        def f(a, b, c, d, e):
-            return a
-
-        expected = f(*args)
-        f_new, res = _argnums_partial(f, args, -1)
+        res = _slice_argnums(args, -1)
         self.assertEqual(res, (4.1,))
-        self.assertEqual(f_new(*res), expected)
 
-        f_new, res = _argnums_partial(f, args, -5)
+        res = _slice_argnums(args, -5)
         self.assertEqual(res, (0.1,))
-        self.assertEqual(f_new(*res), expected)
 
     def test_flat_args_with_tuple_argnum(self):
         args = (0.1, 1.1, 2.1, 3.1, 4.1)
 
-        def f(a, b, c, d, e):
-            return a
-
-        f_new, res = _argnums_partial(f, args, (0, 1, 2, 3, 4))
-        self.assertEqual(f_new(*res), 0.1)
+        res = _slice_argnums(args, (0, 1, 2, 3, 4))
         self.assertEqual(res, args)
 
-        f_new, res = _argnums_partial(f, args, (0, -3))
-        self.assertEqual(f_new(*res), 0.1)
+        res = _slice_argnums(args, (0, -3))
         self.assertEqual(res, (0.1, 2.1))
 
     def test_pytree_args(self):
         args = ((0.1, 1.1), 2.0, [3.1])
 
-        def f(a, b, c):
-            return a[0] + a[1] + b + c[0]
-
-        expected = f(*args)
-
-        f_new, res = _argnums_partial(f, args, 0)
+        res = _slice_argnums(args, 0)
         self.assertEqual(res, args[0:1])
-        self.assertEqual(f_new(*res), expected)
 
-        f_new, res = _argnums_partial(f, args, (0,))
+        res = _slice_argnums(args, (0,))
         self.assertEqual(res, args[0:1])
-        self.assertEqual(f_new(*res), expected)
 
-        f_new, res = _argnums_partial(f, args, -1)
+        res = _slice_argnums(args, -1)
         self.assertEqual(res, args[-1:])
-        self.assertEqual(f_new(*res), expected)
 
-        f_new, res = _argnums_partial(f, args, (0, -2))
+        res = _slice_argnums(args, (0, -2))
         self.assertEqual(res, args[0:2])
-        self.assertEqual(f_new(*res), expected)
 
     def test_argnums_reorders(self):
         args = ((0.1, 1.1, 2.1), 3.1, 4.1)
 
-        def f(a, b, c):
-            return a[0] + a[1] + a[2] + b + c
-
-        expected = f(*args)
-        f_new, res = _argnums_partial(f, args, (1, 0))
+        res = _slice_argnums(args, (1, 0))
         self.assertEqual(res, (args[1], args[0]))
-        self.assertEqual(f_new(*res), expected)
 
     def test_function_with_default_args(self):
         args = ((0.1, 1.1, 2.1), 3.1)
 
-        def f(a, b, c=4.1):
-            return a[0] + a[1] + a[2] + b + c
-
-        expected = f(*args)
-        f_new, res = _argnums_partial(f, args, -2)
+        res = _slice_argnums(args, -2)
```

Comment:
I am not sure `test_function_with_default_args` applies anymore because `_slice_argnums` doesn't accept functions. In exchange for this, could we add a jacfwd/jacrev test with a function that has default kwargs?

Reply:
Good catch! Removed and added the jac test